### ISSUE TITLE
ASM?

### ISSUE THREAD

**Issue by hmyh1202** [guoweilong/cgmaptools#3]
Hi WeiLong:
  I have a question about the ASM.
  What format of VCF info should I provide for ASM.pl, is the Bis-SNP output file  works?  if only heterozygous SNP site used for ASM calling? And how is it works when I don`t have a SNP file？
  And what was advanced of SNV calling, which method should I select.
  I have run it for >24h with no output..

Best Han
2017/11/29

**Comment by guoweilong** [comment]
Hi Han,

Thanks for your question. Dr. Ping Zhu and I made the following reply, and hope you find them satisfied.

1. `asm` works for SNP files in standard VCF format, however, we did not test the output of Bis-SNP yet. If it does not work, you’d better reformat the Bis-SNP results.
2. ASM analysis is absolutely a workflow for heterozygous SNP sites.
3. A SNP file is required for `asm` to run.
4. One of our major advance in CGmapTools, is that we designed BayesWC and BinomWC, improved the precision in heterozygous SNV calling from ~80% to 99%, from simulation data. You can read our published paper for more detail 
Weilong Guo*#, Ping Zhu*, Matteo Pellegrini, Michael Zhang, Xiangfeng Wang, Zhongfu Ni. (2017) [CGmapTools improves the precision of heterozygous SNV calls and supports allele-specific methylation detection and visualization in bisulfite-sequencing data](https://doi.org/10.1093/bioinformatics/btx595), Bioinformatics, btx595.
5. Generally, BayesWC is more accurate and have more information, BinomWC is faster.
6. As you mentioned "run it for >24h with no output", can you send your full commands to us? Are you using ATCGmap file as input? Generally, the SNV calling step should give you output on screen in seconds.

Best,
Weilong and Ping

**Comment by hmyh1202** [comment]
Thans Dr. Ping Zhu and U !

1.  Now I have got the ouput of ASM by using output of BisSNP. I was seems right.

2.  I  will try BayesWC method for SNV calling later.

3.  I did not use ATCGmap file input?  And, my commands was: perl ASM.pl asm -r *fa -b Sample1_P_R1_bismark_bt2_pe.dedupped.bam -l Sample1_P_R1_bismark_bt2_pe.addheader.snp.filtered.sort.vcf (_produced by BisSNP_) -o ASM.tsv
It was consumed nearly 3 days! The bam file contains 116430000 reads pairs.
Now the output of **ASM.tsv** is as below:
Chr	Pos	Ref	Allele1	Allele2	Allele1_linked_C	Allele2_linked_C	Allele1_linked_C_met  Allele2_linked_C_met	pvalue	fdr	ASM
NC_019458.2	7624	A	A	G	1-1-1-0.82-0.47-0.75-0.9-1-1-0.5-1	1-0-1-1-1-1-1-1	0.86  0.88	9.06e-01	1.00e+00	FALSE

 The Best.
